src/shipit_pulse_listener/shipit_pulse_listener/listener.py

```diff
@@ -13,9 +13,91 @@
 logger = get_logger(__name__)
 
 
-class HookStaticAnalysis(Hook):
+class HookPhabricator(Hook):
     '''
     Taskcluster hook handling the static analysis
+    for Phabricator differentials
+    '''
+    def __init__(self, configuration):
+        assert 'hookId' in configuration
+        super().__init__(
+            'project-releng',
+            configuration['hookId'],
+        )
+
+        # Save phabricator connection data
+        assert 'phabricator_url' in configuration
+        assert 'phabricator_token' in configuration
+        self.api_url = configuration['phabricator_url']
+        self.api_token = configuration['phabricator_token']
+
+        # Start on first page
+        self.before = None
+
+    def list_differential(self):
+        '''
+        List new differential items
+        '''
+        url = '{}/differential.revision.search'.format(self.api_url)
+        after = None
+        while True:
+            logger.info('Loading phabricator differentials', after=after, before=self.before)
+            payload = {
+                'api.token': self.api_token,
+                'after': after,
+                'limit': 20,
+            }
+            if after is None and self.before:
+                # Initial page
+                payload['before'] = self.before
+            response = requests.post(url, payload)
+            response.raise_for_status()
+            data = response.json()
+            assert data['error_code'] is None, \
+                'Conduit error: {} - {}'.format(
+                    data['error_code'],
+                    data['error_info'],
+                )
+
+            # Pass results
+            diffs = data['result']['data']
+            for diff in diffs:
+                yield diff
+
+            # Save first id for future executions
+            if after is None and diffs:
+                self.before = diffs[0]['id']
+
+            # Load next page
+            after = data['result']['cursor']['after']
+            if after is None:
+                break
+
+    async def build_consumer(self, *args, **kwargs):
+        '''
+        Query phabricator differentials regularly
+        '''
+        while True:
+
+            # Get new differential ids
+            for diff in self.list_differential():
+                if diff['type'] != 'DREV':
+                    logger.info('Skipping differential', id=diff['id'], type=diff['type'])
+                    continue
+
+                # Create new task
+                await self.create_task({
+                    'PHABRICATOR': '{id}:{phid}'.format(**diff)
+                })
+
+            # Sleep a bit before trying new diffs
+            await asyncio.sleep(60)
```

Comment:
Would it make sense to be able to configure polling delays for Differentials and Pulse messages?

Reply:
There is no delay for Pulse message, they are processed in a Pub-Sub manner.
I hardcoded the differential delay for now, as i'm not sure yet this is the best way to get updates... That will do for now i guess

Reply:
Ok, works for me. A `FIXME` comment would be great here.

Reply:
Issue #730